packages/vscode-extension/src/browser/extension.ts

```diff
@@ -0,0 +1,117 @@
+  // const formattingProvider = languages.registerDocumentFormattingEditProvider(
+  //   LIQUID,
+  //   new LiquidFormatter(diagnosticCollection, diagnosticTextDocumentVersion),
+  // );
+  // context.subscriptions.push(formattingProvider);
+
+  // workspace.onDidChangeConfiguration(onConfigChange(context));
+
+  // // If you change the file, the prettier syntax error is no longer valid
+  // workspace.onDidChangeTextDocument(({ document }) => {
+  //   const bufferVersionOfDiagnostic = diagnosticTextDocumentVersion.get(document.uri);
+  //   if (bufferVersionOfDiagnostic !== document.version) {
+  //     diagnosticCollection.delete(document.uri);
+  //   }
+  // });
+
+  await startServer(context);
+}
+
+export function deactivate() {
+  return stopServer();
+}
+
+async function startServer(context: ExtensionContext) {
+  console.log('Starting Theme Check Language Server');
+  const clientOptions: LanguageClientOptions = {
+    documentSelector: [
+      { scheme: 'file', language: 'liquid' },
+      { scheme: 'file', language: 'plaintext' },
+      { scheme: 'file', language: 'html' },
+      { scheme: 'file', language: 'javascript' },
+      { scheme: 'file', language: 'css' },
+      { scheme: 'file', language: 'scss' },
+      { scheme: 'file', language: 'json' },
+      { scheme: 'file', language: 'jsonc' },
+    ],
+  };
+
+  client = createWorkerLanguageClient(context, clientOptions);
+
+  client.start();
+  console.log('Theme Check Language Server started');
+}
+
+async function stopServer() {
+  try {
+    if (client) {
+      await Promise.race([client.stop(), sleep(1000)]);
+    }
+  } catch (e) {
+    console.error(e);
+  } finally {
+    client = undefined;
+  }
+}
+
+async function restartServer(context: ExtensionContext) {
+  if (client) {
+    await stopServer();
+  }
+  await startServer(context);
+}
+
+const onConfigChange =
+  (context: ExtensionContext) => (event: { affectsConfiguration: (arg0: string) => any }) => {
+    const didChangeThemeCheck = event.affectsConfiguration('shopifyLiquid.languageServerPath');
+    const didChangeShopifyCLI = event.affectsConfiguration('shopifyLiquid.shopifyCLIPath');
+    const didChangeLegacyMode = event.affectsConfiguration('shopifyLiquid.legacyMode');
+    if (didChangeThemeCheck || didChangeShopifyCLI || didChangeLegacyMode) {
+      restartServer(context);
+    }
+  };
+
+function createWorkerLanguageClient(
+  context: ExtensionContext,
+  clientOptions: LanguageClientOptions,
+) {
+  // Create a worker. The worker main file implements the language server.
+  const serverMain = Uri.joinPath(context.extensionUri, 'dist', 'browser', 'server.js');
+  const worker = new Worker(serverMain.toString(true));
+
+  // create the language server client to communicate with the server running in the worker
+  return new LanguageClient(
+    'shopifyLiquid',
+    'Theme Check Language Server',
+    clientOptions,
+    worker,
+  );
+}
```

Comment:
Major difference is that here we start the language server in a Web Worker instead of settingup STDIO and a subprocess.